src/main/java/com/exasol/exaoperation/plugin/Plugin.java

```diff
@@ -0,0 +1,157 @@
+package com.exasol.exaoperation.plugin;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.Container.ExecResult;
+
+import com.exasol.exaoperation.ExaOperationEmulatorException;
+
+/**
+ * Representation of an EXAoperation plug-in.
+ * <p>
+ * These plug-ins are installed from a plug-in package, an archive file with a defined structure and contain scripts or
+ * executables called in the various life cycle phases of the plug-in (e.g. for installation, start and stop).
+ * </p>
+ */
+public class Plugin {
+    public static final String PLUGIN_PACKAGE_PREFIX = "Plugin.";
+    private static final Logger LOGGER = LoggerFactory.getLogger(Plugin.class);
+    @SuppressWarnings("squid:S4784") // This is a test framework RegEx DoS attacks are unrealistic since this would mean
+                                     // the testers are attacking themselves.
+    private static final Pattern PLUGIN_PACKAGE_FILENAME_PATTERN = Pattern
+            .compile("Plugin\\.([.\\w]+-\\d+(?:\\.\\d+)*).*?");
```

Comment:
It will accept `Plugin.Foo-1-0-0.pkg` as a valid match, but maybe it should fail? The extracted name would be `Plugin.Foo-1`.

Reply:
You are right. This is intended behavior.